### DISCUSSION TITLE
Document how to config image registry

### DISCUSSION THREAD

**Discussion by guoyingc** [vhive-serverless/vHive#376] [Q&A] [answered]
Hi, vhive team

I would like to try vhive but I met with issues to store function (knative service) images. I tried several ways:
- docker hub: if I use docker hub, I will meet with "Too Many Requests - Server message: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading" error when pod is starting
- local registry: if I use local registry, I will meet with "http: server gave HTTP response to HTTPS client" when pod is starting

Could you enhance the document and add guidance to config the registry for function images? Either how to config the credential of image registry or config http access to image registry ? I've been stuck in this issue and search the internet for a whole day but I cannot find a way to config the image registry for firecracker-containerd. I tried [containerd Configure Image Registry](https://github.com/containerd/containerd/blob/main/docs/cri/registry.md) but it doesn't work.

Thanks.

**Comment by ustiugov** [comment, marked as answer]
Converted to a discussion. Indeed, DockerHub have become stricter, allowing fewer pull requests in a time period. My recommendation is to host your images in the [GitHub registry](ghcr.io). Public images do not have limitations there.

Let me know if this helps.

**Reply by guoyingc** [reply]
Thanks. I will have a try.
